### PR TITLE
[terra-application] Workspace Title Override

### LINKED ISSUE #161: [terra-application] New workspace tab title API override
# Feature Request
UX has requested the ability to provide further descriptive text that can be placed within the visible tabs title section.

## Description
The requested behavior would allow consumers to provide a new prop that would override the default value of the tab title, the title itself, to be more descriptive, in order reduce the confusion of duplicate text. This newly provided text should be representative of the tab's title, but with more context.

## @ Mentions
@neilpfeiffer @ryanthemanuel @mjhenkes @tbiethman

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Add property for workspace title

Closes #161 

### Deployment Link

https://terra-applic-workspace--ahgvxf.herokuapp.com/tests/terra-application/workspace/workspace

### Testing
Added jest test.

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by tbiethman — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Review by benbcai — APPROVED** [review]

File: src/workspace/WorkspaceContent.jsx [hunk 1]
```diff
@@ -17,6 +17,10 @@ const propTypes = {
    * Child node content to be displayed within the content region.
    */
   children: PropTypes.node,
+  /**
+   * Text to be displayed as the title of the workspace content.
```
Comment (neilpfeiffer) [inline src/workspace/WorkspaceContent.jsx:21]:
```suggestion
   * Text to be displayed as the title of the workspace content.
   * Special Note: this prop is optional and should be used with caution. If this prop is not provided, the workspace content title is injected by default using the label specified in `WorkspaceItem` _(recommended without additional guidance)_. Providing this prop will override the default text and will need to follow proper accessibility guidelines.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 ## Unreleased
+* Added
+  * Added `label` prop to WorkspaceContent to allow label override.
```
Comment (neilpfeiffer) [inline CHANGELOG.md]:
```suggestion
  * Added optional `label` prop to WorkspaceContent to allow label override for guided usage.
```

File: src/workspace/WorkspaceContent.jsx (hunk 1 above)
Comment (neilpfeiffer) [reply src/workspace/WorkspaceContent.jsx:21]: Thinking that this special note can be moved to a standout/notice on a future PR when we provide full documentation. We then can replace this with something like _"See [xyz...] for usage instructions."_ at that time.

**Review by neilpfeiffer — APPROVED** [review]
Reviewed: #162
Resolves: #161

UX +1

**Pull request merged**